src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Comment.cs

```diff
@@ -71,115 +60,96 @@ private void WriteCommentByOptions(ReadOnlySpan<char> value)
+            Debug.Assert(status != OperationStatus.DestinationTooSmall);
+            BytesPending += written;
+
+            output[BytesPending++] = JsonConstants.Asterisk;
+            output[BytesPending++] = JsonConstants.Slash;
         }
 
-        private void WriteCommentEscapeValue(ReadOnlySpan<char> value, int firstEscapeIndexVal)
+        private void WriteCommentIndented(ReadOnlySpan<char> value)
         {
-            Debug.Assert(int.MaxValue / JsonConstants.MaxExpansionFactorWhileEscaping >= value.Length);
-            Debug.Assert(firstEscapeIndexVal >= 0 && firstEscapeIndexVal < value.Length);
+            int indent = Indentation;
+            Debug.Assert(indent <= 2 * JsonConstants.MaxWriterDepth);
 
-            char[] valueArray = null;
+            Debug.Assert(value.Length < (int.MaxValue / JsonConstants.MaxExpansionFactorWhileTranscoding) - indent - 4 - s_newLineLength);
 
-            int length = JsonWriterHelper.GetMaxEscapedLength(value.Length, firstEscapeIndexVal);
+            // All ASCII, /*...*/ => escapedValue.Length + 4
+            // Optionally, 1-2 bytes for new line, and up to 3x growth when transcoding
+            int maxRequired = indent + (value.Length * JsonConstants.MaxExpansionFactorWhileTranscoding) + 4 + s_newLineLength;
 
-            Span<char> escapedValue;
-            if (length > StackallocThreshold)
-            {
-                valueArray = ArrayPool<char>.Shared.Rent(length);
-                escapedValue = valueArray;
-            }
-            else
+            if (_memory.Length - BytesPending < maxRequired)
             {
-                // Cannot create a span directly since it gets passed to instance methods on a ref struct.
-                unsafe
-                {
-                    char* ptr = stackalloc char[length];
-                    escapedValue = new Span<char>(ptr, length);
-                }
+                Grow(maxRequired);
             }
-            JsonWriterHelper.EscapeString(value, escapedValue, firstEscapeIndexVal, out int written);
 
-            WriteCommentByOptions(escapedValue.Slice(0, written));
+            Span<byte> output = _memory.Span;
 
-            if (valueArray != null)
+            if (_tokenType != JsonTokenType.None)
             {
-                ArrayPool<char>.Shared.Return(valueArray);
+                WriteNewLine(output);
             }
+
+            JsonWriterHelper.WriteIndentation(output.Slice(BytesPending), indent);
+            BytesPending += indent;
+
+            output[BytesPending++] = JsonConstants.Slash;
+            output[BytesPending++] = JsonConstants.Asterisk;
+
+            ReadOnlySpan<byte> byteSpan = MemoryMarshal.AsBytes(value);
+            OperationStatus status = JsonWriterHelper.ToUtf8(byteSpan, output.Slice(BytesPending), out int _, out int written);
+            Debug.Assert(status != OperationStatus.DestinationTooSmall);
```

Comment:
Note: This could be invalid data or needs more data, and we just stop writing at this point. There is no recovery/replacement.